matching-core/src/main/java/com/graphhopper/matching/EdgeMatch.java

```diff
@@ -28,9 +30,11 @@
 
     private final EdgeIteratorState edgeState;
     private final List<GPXExtension> gpxExtensions;
+    private final PointList wayGeometry;
 
-    public EdgeMatch(EdgeIteratorState edgeState, List<GPXExtension> gpxExtension) {
+    public EdgeMatch(EdgeIteratorState edgeState, List<GPXExtension> gpxExtension, PointList wayGeometry) {
         this.edgeState = edgeState;
+        this.wayGeometry = wayGeometry;
```

Comment:
Hmmh, I would prefer a setWayGeometry, this way we can init the EdgeMatch and call setWayGeometry for the certain conditions. Which will make "Match Phase (4)" a little less indirect - i.e. we can avoid `PointList wayGeometry = null;`